sktime/base/_base.py

```diff
@@ -14,14 +54,160 @@
 from sktime.exceptions import NotFittedError
 
 
-class BaseEstimator(_BaseEstimator):
+class BaseObject(_BaseEstimator):
+    """Base class for parametric objects with tags sktime.
+
+    Extends scikit-learn's BaseEstimator to include sktime interface for tags.
+    """
+
+    def __init__(self):
+        self._tags_dynamic = dict()
```

Comment:
If we name the others "class tags", we should name these "object tags" 
```suggestion
        self._object_tags = dict()
```

Reply:
hm, I'd prefer not to, conceptually I still feel it's more dynamic overrides of class tags rather than tags that are specifically object tags, because conceptually the class corresponds to the algorithm.

I prefer to think about the dynamic tags as "conditional class tags", i.e., specifying a class tag which is a function of component class tags.

Reply:
I think the object/class or static/dynamic distinction makes sense, we can explain details in the docs, but when we call one "class_tags" and the other "dynamic_tags" their relation isn't even clear

Reply:
the reference to dynamic only happens internally, users never see this.

The general terminology is "tags" and "class tags". Introducing any dichotomy will make the user think there are two sets of tags. Instead they should call `get_tags` primarily, this has *no qualifier* and will retrieve all tags (both object/class, or both static/dynamic).

So I think the current naming makes sense, except perhaps the internal variables which neither users nor extenders should ever access, only core developers working specifically with `BaseObject`.

Reply:
I know, but for core devs it would still make the relation clearer between the two, but it's a minor point

Reply:
I have no strong feeling, but I think it's clearer that way - implicit dichotomies are strong in (mis)guiding perception, even (and especially) if they are fake